occ_poll_app_handler.cpp

```diff
@@ -0,0 +1,556 @@
+                {
+                    lg2::error("OccPollAppHandler::PushPowrSensorsToDbus: functionID({FXID}) unsuported power sensor name", "FXID", functionID);
+                    TraceOncePwrFuncId = false;
+                }
+            }
+            index += bytesPerSensor;
+        }
+    }
+    else
+    {
+        if (TraceOncePollSensor)
+        {
+            lg2::error("OccPollAppHandler::PushPowrSensorsToDbus: unsuported format:{BYTE}",
+                "BYTE", lg2::hex, SensorFormat);
+            TraceOncePollSensor = false;
+        }
+
+        index += bytesPerSensor * NumberSensors;
+    }
+
+}//end PushPowrSensorsToDbus
+
+void OccPollAppHandler::PushCapsSensorsToDbus(uint16_t& index )
+{
+    // Caps Sensor Header Data format for POLL
+    //  SensorFormat:   1 byte
+    //  bytesPerSensor: 1 byte
+    //  NumberSensors:  1 byte
+    //----------------------
+
+    uint8_t SensorFormat = PollRspData[index++];
+    uint8_t bytesPerSensor = PollRspData[index++];
+    uint8_t NumberSensors = PollRspData[index++];
+
+    if (SensorFormat == CAPS_SENSOR_FORMAT)
+    {
+        for (uint16_t i = 0; i < NumberSensors; i++)
+        {
+            // Caps Sensor Record Data format for POLL
+            //  SensorCap:       2 byte
+            //  PollRspMaxPower: 2 byte
+            //  NCap:            2 byte
+            //  MaxCap:          2 byte
+            //  HardMinCap:      2 byte
+            //  SoftMinCap:      2 byte
+            //  UserCap:         2 byte
+            //  UserSourse:      1 byte
+            //----------------------
+
+            PollRspMaxPower = (static_cast<uint16_t>(PollRspData[index + 2]) << 8) |
+                    (static_cast<uint16_t>(PollRspData[index + 3]));
+
+            std::string sensorPath = OCC_SENSORS_ROOT + std::string("/power/");
+            sensorPath.append("total_power");
+
+            dbus::OccDBusSensors::getOccDBus().setUnit(sensorPath, "xyz.openbmc_project.Sensor.Value.Unit.Watts");
+            dbus::OccDBusSensors::getOccDBus().setValue(sensorPath, PollRspMaxPower);
+            dbus::OccDBusSensors::getOccDBus().setOperationalStatus(sensorPath, true);
+            if (statusObject.existingSensors.find(sensorPath) == statusObject.existingSensors.end())
+            {
+                dbus::OccDBusSensors::getOccDBus().setPurpose(sensorPath,"xyz.openbmc_project.Sensor.Purpose.SensorPurpose.TotalPower");
+                dbus::OccDBusSensors::getOccDBus().setChassisAssociation(
+                    sensorPath, {"all_sensors"});
+            }
+            statusObject.existingSensors[sensorPath] = occInstanceID;
+
+
+            PollRspHardMin = (static_cast<uint32_t>(PollRspData[index + 8]) << 8) |
+                    (static_cast<uint32_t>(PollRspData[index + 9]));
+
+            PollRspSoftMin = (static_cast<uint32_t>(PollRspData[index + 10]) << 8) |
+                    (static_cast<uint32_t>(PollRspData[index + 11]));
+
```

Comment:
You also need to save the PollRspMaxPower (different from what you did above which was the current reading)